packages/engine/src/interaction/components/InteractableComponent.ts

```diff
@@ -145,17 +180,33 @@ export const InteractableComponent = defineComponent({
     const entity = useEntityContext()
     const interactable = useComponent(entity, InteractableComponent)
     const input = useOptionalComponent(entity, InputComponent)
+    const isEditing = useMutableState(EngineState).isEditing
+
+    useEffect(() => {
+      setComponent(entity, DistanceFromCameraComponent)
+      setComponent(entity, DistanceFromLocalClientComponent)
+
+      addInteractableUI(entity)
+    }, [])
+
+    useEffect(() => {
+      if (!isEditing.value) {
+        addInteractableUI(entity)
+      } else {
+        removeInteractableUI(entity)
+      }
+    }, [isEditing.value])
 
     useEffect(() => {
-      if (getState(EngineState).isEditor || !input) return
+      if (isEditing.value || !input) return
       const canvas = getComponent(Engine.instance.viewerEntity, RendererComponent).canvas
       if (input.inputSources.length > 0) {
         canvas.style.cursor = 'pointer'
       }
       return () => {
         canvas.style.cursor = 'auto'
       }
-    }, [input?.inputSources.length])
+    }, [input?.inputSources.length, isEditing.value])
```

Comment:
isEditing should not be unwrapped with .value, just use the actual state (in this case just isEditing) for reactor dependency arrays.

Reply:
either is fine actually if the effect is reversable

Reply:
.value will re-render if the value changes, not .value will rerender any time a .set() is called on the state (even if the value doesnt change)